x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -70,24 +75,24 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
     this.pendingCount$.next(this.pendingCount$.getValue() + 1);
 
     return this.runSearch(request, combinedSignal, strategy).pipe(
-      expand((response) => {
+      tap((r) => {
         // If the response indicates of an error, stop polling and complete the observable
-        if (isErrorResponse(response)) {
+        if (isErrorResponse(r)) {
           return throwError(new AbortError());
         }
-
-        // If the response indicates it is complete, stop polling and complete the observable
-        if (isCompleteResponse(response)) {
-          return EMPTY;
+      }),
+      concatMap((r) => {
+        if (isCompleteResponse(r)) {
+          return of(r);
         }
 
-        id = response.id;
-        // Delay by the given poll interval
-        return timer(pollInterval).pipe(
-          // Send future requests using just the ID from the response
-          mergeMap(() => {
-            return this.runSearch({ ...request, id }, combinedSignal, strategy);
-          })
+        return merge(
+          of(r),
+          takePartialSearch(
```

Comment:
I think you had to use `takePartialSearch` here instead of `doPartialSearch` once again, because other places rely on `config$` to start the stream. 

I think we could adjust this so that we use `doPartialSearch` here too.